drivers/pwm/pwm_stm32.c

```diff
@@ -785,6 +797,7 @@ static void pwm_stm32_irq_config_func_##index(const struct device *dev)		\
 		.prescaler = DT_PROP(PWM(index), st_prescaler),		       \
 		.countermode = DT_PROP(PWM(index), st_countermode),	       \
 		.deadtime = DT_PROP(PWM(index), st_deadtime),		       \
+		.mastermode = CONCAT(LL_TIM_TRGO_, DT_STRING_TOKEN(PWM(index), st_mastermode)),	\
```

Comment:
I see there is 1 SoC series for which `LL_TIM_TRGO_RESET` is not defined and that supports `"st,stm32-pwm"` compatible driver: STM32WB0.
```suggestion
#ifdef LL_TIM_TRGO_RESET
#define STM32_LL_HAS_TIM_TRGO_RESET 1
#else
#define STM32_LL_HAS_TIM_TRGO_RESET 0
#endif

(...)

		.mastermode = COND_CODE_1(STM32_LL_HAS_TIM_TRGO_RESET,	       \
					  CONCAT(LL_TIM_TRGO_,		       \
						 DT_STRING_TOKEN(PWM(index),   \
						 st_mastermode)),	       \
					  (0)),				       \
```
or like.